### ISSUE TITLE
Wallet.extended_key  ,  `public_key_hex` and `public_key_base58` is wrong

### ISSUE THREAD

**Issue by youngqqcn** [meherett/eth-wallet#3]
```python
def extended_key(self, private_key=True, encoded=True):
    version = EXTEND_MAIN_PRIVATE[0] \
        if private_key else EXTEND_MAIN_PUBLIC[0]
    depth = bytes(bytearray([self.depth]))
    parent_fingerprint = self.parent_fingerprint
    child = struct.pack(">L", self.index)
    chain = self.chain
    data = b"\x00" + unhexlify(self.private_key())  ######### it's always use self.private key 
    try:
        raw = (version + depth +
               parent_fingerprint + child + chain + data)
        if encoded:
            return check_encode(raw)
        else:
            return raw.hex()
    except TypeError:
        return None
```

fixed
> reference code:  https://github.com/richardkiss/pycoin/blob/58bac241d4da237249da929192b8ba708bbe4d96/pycoin/key/BIP32Node.py#L95

```python
def extended_key(self, private_key=True, encoded=True):
    version = EXTEND_MAIN_PRIVATE[0] \
        if private_key else EXTEND_MAIN_PUBLIC[0]
    depth = bytes(bytearray([self.depth]))
    parent_fingerprint = self.parent_fingerprint
    child = struct.pack(">L", self.index)
    chain = self.chain
    
    ###### fixed #####
    data = b"\x00" + unhexlify(self.private_key()) if private_key else unhexlify(self.public_key())  
    ###################
    try:
        raw = (version + depth +
               parent_fingerprint + child + chain + data)
        if encoded:
            return check_encode(raw)
        else:
            return raw.hex()
    except TypeError:
        return None
```

## Test

> https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#test-vector-1

```python
def test_from_seed_1():

    wallet = Wallet()


    wallet.from_seed(seed='000102030405060708090a0b0c0d0e0f')
    # wallet.from_index(index=1)
    print( json.dumps( wallet.dumps() , indent=4) )

    # reference:   https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#test-vector-1
    xpub = 'xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8'
    xpriv = 'xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKmPGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi'

    assert xpriv ==  wallet.extended_key(private_key=True, encoded=True)
    assert xpub == wallet.extended_key(private_key=False, encoded=True)

    pass
```

**Comment by meherett** [comment]
Fixed complete at [v0.2.7](https://github.com/meherett/eth-wallet/releases/tag/v0.2.7)

**Comment by meherett** [comment]
Fixed complete at [v0.2.7](https://github.com/meherett/eth-wallet/releases/tag/v0.2.7)
